tests/test-Image.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+/**
+ * Class A8C_Files_Image_Test
+ *
+ * @group srcset
+ */
+class A8C_Files_Image_Test extends \WP_UnitTestCase {
+
+	/**
+	 * The test image.
+	 *
+	 * @var string
+	 */
+	public $test_image = __DIR__ . '/fixtures/image.jpg'; //@todo: consider using `DIR_TESTDATA . '/images/canola.jpg';`
+
+	/**
+	 * Load the A8C_Files\ImageSizes class.
+	 */
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+
+		require_once( __DIR__ . '/../a8c-files/Image.php' );
+	}
+
+	/**
+	 * Set the test to the original initial state of the VIP Go.
+	 *
+	 * 1. A8C files being in place, no srcset.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->enable_a8c_files();
+	}
+
+	/**
+	 * Cleanup after a test.
+	 *
+	 * Remove added uploads.
+	 */
+	public function tearDown() {
+
+		$this->remove_added_uploads();
+
+		parent::tearDown();
+	}
+
+	/**
+	 * Helper function for turning the a8c_files implementation on.
+	 */
+	public function enable_a8c_files() {
+		add_action( 'init', 'a8c_files_init' );
+		add_filter( 'intermediate_image_sizes', 'wpcom_intermediate_sizes' );
+		add_filter( 'intermediate_image_sizes_advanced', 'wpcom_intermediate_sizes' );
+	}
+
+	/**
+	 * Helper function for accessing protected method.
+	 *
+	 * @param string $name Name of the method.
+	 *
+	 * @return ReflectionMethod
+	 */
+	protected static function getMethod( $name ) {
+		$class = new ReflectionClass( 'A8C_Files\\Image' );
+		$method = $class->getMethod( $name );
+		$method->setAccessible(true);
+		return $method;
+	}
+
+	/**
+	 * Helper function for accessing protected property.
+	 *
+	 * @param string $name Name of the property.
+	 *
+	 * @return ReflectionProperty
+	 */
+	protected function getProperty( $name ) {
+		$class = new ReflectionClass( 'A8C_Files\\Image' );
+		$property = $class->getProperty( $name );
+		$property->setAccessible(true);
+		return $property;
+	}
+
+	/**
+	 * Unit test covering the object initialisation.
+	 *
+	 * @covers A8C_Files\Image::__construct
+	 *
+	 * @param array $expected_sizes Array of expected sizes.
+	 */
+	public function test__object_construction() {
+
+		$attachment_id = self::factory()->attachment->create_object(
+			$this->test_image, 0, [
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			]
+		);
+
+		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+		$postmeta = get_post_meta( $attachment_id, '_wp_attachment_metadata', true );
+
+		$image = new A8C_Files\Image( $postmeta, $attachment_id );
+
+		$this->assertEquals( $postmeta['width'], $image->get_width() );
+		$this->assertEquals( $postmeta['height'], $image->get_height() );
+		$this->assertEquals( 'image/jpeg', $image->get_mime_type() );
+		$this->assertFalse( $image->is_resized() );
```

Comment:
Minor: Any time there are multiple assert calls within a single test, it helps to include the `message` param to help identify which assertion failed. Otherwise, phpunit will report something like `Failed asserting that 125 matches expected 100`, which doesn't really give us a clear idea of what failed.